en/middleware/micrortps.md

```diff
@@ -0,0 +1,271 @@
+
+The source code for the *Client* and *Agent* will be generated in **build\__BUILDPLATFORM_/src/modules/micrortps_bridge**, under the **/micrortps_client** and **/micrortps_agent** folders (respectively).
+
+
+### Supported uORB/RTSP messages
+
+The set of uORB topics that can be sent and received by the bridge are listed in the **.cmake** file (**cmake/configs**) for each target platform:
+
+```cmake
+set(config_rtps_send_topics
+  sensor_combined
+   # Add new topic...
+   )
+
+set(config_rtps_receive_topics
+   vehicle_command
+   # Add new topic...
+   )
+```
+
+A uORB serialization function will be generated for each topic in both lists. An IDL file will also be created for each topic from the associated uORB **.msg** files 
+
+> **Note** FastRTSP uses IDL files to define the structure of RTPS messages (in this case, RTPS messages for sending uORB topics). The IDL files are compiled to C++ by the *fastrtpsgen* tool, which is then used by the Agent and other FastRTPS applications.
+
+
+#### uORB serialization code
+
+Serialization functions are generated for the uORB topics that are to be sent/received. 
+For example, the following functions would be generated for the *sensor_combined.msg*:
+
+```sh
+void serialize_sensor_combined(const struct sensor_combined_s *input, char *output, uint32_t *length, struct microCDR *microCDRWriter);
+void deserialize_sensor_combined(struct sensor_combined_s *output, char *input, struct microCDR *microCDRReader);
+```
+
+
+## Client (PX4 Firmware)
+
+The *Client* source code is generated, and the executable is automatically built and included when you make the firmware (in the normal way). For example, to build and upload the firmware:
+
+For NuttX/Pixhawk flight controllers:
+```sh
+make px4fmu-v4_default upload
+```
+For Qualcomm Snapdragon Flight:
+```sh
+$ make eagle_default upload
+  ```
+
+After uploading the firmware, the *Client* application can be launched by typing its name and passing a variable number of arguments as shown below:
+
+```sh
+> micrortps_client start|stop [options]
+  -t <transport>          [UART|UDP] Default UART
+  -d <device>             UART device. Default /dev/ttyACM0
+  -u <update_time_ms>     Time in ms for uORB subscribed topics update. Default 0
+  -l <loops>              How many iterations will this program have. -1 for infinite. Default 10000
+  -w <sleep_time_ms>      Time in ms for which each iteration sleep. Default 1ms
+  -b <baudrate>           UART device baudrate. Default 460800
+  -p <poll_ms>            Time in ms to poll over UART. Default 1ms
+  -r <reception port>     UDP port for receiving. Default 2019
+  -s <sending port>       UDP port for sending. Default 2020
+```
+
+> **Note** If the selected UART port is busy, it's possible that the MAVLink application is already being used. If it is the case, you can stop MAVLink from NuttShell by typing:
```

Comment:
@hamishwillee @VicenteEprosima It can be either of them.
1. In this case you probably selected the wrong port for either the bridge or mavlink. So stopping mavlink is good enough for testing, but is not a real solution
2. In this case some documentation is needed for how to run the bridge and mavlink on the same port.